a_test.go

```diff
@@ -183,42 +186,360 @@ func TestA_uses_Logger_dynamic_interface(t *testing.T) {
+	var ctxKey ctxKeyT
+
+	cases := []struct {
+		name     string
+		parallel bool
+	}{
+		{
+			name:     "parallel",
+			parallel: true,
+		},
+		{
+			name: "sequential",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			flow, task, childTasksResults, loggerSpy := prepareFlowAndTask(t, 3, ctxKey, func(a *goyek.A, i int) {
+				a.Skip(i)
+
+				a.Cleanup(func() {
+					skipped := a.Skipped()
+					assertTrue(t, skipped, "a.Skipped() should return true for "+a.Name())
+				})
+			})
+			childTasks := prepareTasks(t, flow, len(childTasksResults), c.parallel, task, func(i int) func(a *goyek.A) {
+				return func(a *goyek.A) {
+					a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+					call := onceCall(t, a.Name())
+
+					call()
+					prevCtx := a.Context()
+					newA := a.WithContext(context.WithValue(prevCtx, ctxKey, i))
+					assertEqual(t, newA.Name(), a.Name(), "name changed for "+a.Name())
+					assertEqual(t, a.Context(), prevCtx, "context changed for "+a.Name())
+
+					task.Action()(newA)
+
+					call()
+					a.Cleanup(func() {
+						skipped := a.Skipped()
+						assertTrue(t, skipped, "a.Skipped() should return true for "+a.Name())
+					})
+				}
+			})
+
+			main := flow.Define(goyek.Task{
+				Name: "main",
+				Deps: childTasks,
+			})
+
+			err := flow.Execute(context.Background(), []string{main.Name()})
+			assertPass(t, err, "flow execute")
+
+			for index, result := range childTasksResults {
+				assertTrue(t, result, "child result "+strconv.Itoa(index)+" is incorrect")
+			}
+
+			assertTrue(t, loggerSpy.called, "logger call")
+			assertEqual(t, loggerSpy.skipCallCount, len(childTasks), "skip calls count")
+		})
+	}
+}
+
+func prepareFlowAndTask(t *testing.T, childCount int, ctxKey interface{}, additionalAction func(*goyek.A, int)) (*goyek.Flow, *goyek.DefinedTask, []bool, *helperLoggerSpy) {
+	t.Helper()
+
+	flow := &goyek.Flow{}
+
+	flow.SetOutput(io.Discard)
+	loggerSpy := &helperLoggerSpy{}
+	flow.SetLogger(loggerSpy)
+
+	depsTask := flow.Define(goyek.Task{
+		Name: "deps",
+		Action: func(a *goyek.A) {
+			a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+		},
+	})
+
+	childTasksResults := make([]bool, childCount)
+	return flow, flow.Define(goyek.Task{
+		Name: "task",
+		Action: func(a *goyek.A) {
+			a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+
+			k, ok := a.Context().Value(ctxKey).(int)
+			if !ok {
+				return
+			}
+
+			childTasksResults[k] = true
+
+			additionalAction(a, k)
+		},
+		Deps: goyek.Deps{depsTask},
+	}), childTasksResults, loggerSpy
+}
+
+func prepareTasks(t *testing.T, flow *goyek.Flow, count int, parallel bool, src *goyek.DefinedTask, actionBuilder func(i int) func(a *goyek.A)) []*goyek.DefinedTask {
+	t.Helper()
+
+	childTasks := make(goyek.Deps, count)
+	for i := 0; i < count; i++ {
+		i := i
+		childTasks[i] = flow.Define(goyek.Task{
+			Name:     "child task " + strconv.Itoa(i),
+			Action:   actionBuilder(i),
+			Deps:     goyek.Deps{src},
+			Parallel: parallel,
```

Comment:
Does this functionality have anything to do with `Parallel` execution?
The `a` with modified context are passed via callstack.
I think we can remove all `parallel` tests as we are testing existing functionality of parallel invocation and not how `WithContext` behaves.

Reply:
I think it should be left in to make sure it doesn't cause any problems, there are no unnecessary tests at all. The main problem I had was with parallel=true, and I think it should be left in.